bot/define_command.go

```diff
@@ -0,0 +1,58 @@
+package bot
+
+import (
+	"fmt"
+	"io"
+	"net/url"
+	"strings"
+
+	"github.com/urfave/cli"
+	"github.com/zpatrick/rclient"
+)
+
+// DatamuseAPIEndpoint for the datamuse api
+const DatamuseAPIEndpoint = "https://api.datamuse.com"
+
+// DatamuseResponse is the response type for a Tenor API search
+type DatamuseResponse []struct {
+	Definitions []string `json:"defs"`
+	Word        string   `json:"word"`
+}
+
+// NewDefineCommand returns a cli.Command that manages !gif
+func NewDefineCommand(endpoint string, w io.Writer) cli.Command {
+	client := rclient.NewRestClient(endpoint)
+	return cli.Command{
+		Name:        "!define",
+		Usage:       "display a definition for query",
+		Description: "A list of definitions is return with a preface of n for nouns or v is for verbs",
+		ArgsUsage:   "args...",
+		Action: func(c *cli.Context) error {
+			args := c.Args()
+			if len(args) == 0 {
+				return fmt.Errorf("At least one argument is required")
+			}
+
+			query := url.Values{}
+			query.Set("sp", strings.Join(args, " "))
+			query.Set("max", "1")
+			query.Set("md", "d")
+
+			var response DatamuseResponse
+			if err := client.Get("/words", &response, rclient.Query(query)); err != nil {
+				return err
+			}
+
+			if len(response) == 0 {
+				return fmt.Errorf("No definitions matching query '%s'", query.Get("sp"))
+			}
+
+			text := fmt.Sprintf("Here are the defintions for %s: \n", response[0].Word)
+			for i := 0; i < len(response[0].Definitions) && i != 5; i++ {
```

Comment:
I'm kind of on the fence here - seems like this could be too verbose; maybe just do the first definition?

Reply:
Maybe the first 3? or allow some sort of expand or we could do count again.